src/m365/spo/commands/site/site-recyclebinitem-remove.ts

```diff
@@ -0,0 +1,167 @@
+import { Cli } from '../../../../cli/Cli';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request, { CliRequestOptions } from '../../../../request';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  ids?: string;
+  confirm?: boolean;
+}
+
+class SpoSiteRecycleBinItemRemoveCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Permanently deletes specific items from the site recycle bin';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        ids: typeof args.options.ids !== 'undefined',
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '-i, --ids [ids]'
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.ids && !validation.isValidGuidArray(args.options.ids.split(','))) {
+          return 'ids contains invalid GUID';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Permanently deleting specific items from the site recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    if (args.options.confirm) {
+      await this.removeRecycleBinItem(args, logger);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: 'Are you sure you want to permanently delete the items from the site recycle bin?'
+      });
+
+      if (result.continue) {
+        await this.removeRecycleBinItem(args, logger);
+      }
+    }
+  }
+
+  private async removeRecycleBinItem(args: CommandArgs, logger: Logger): Promise<void> {
+    try {
+      await this.postBatch(args.options.ids!.split(','), logger, args.options.siteUrl);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async postBatch(ids: string[], logger: Logger, siteUrl: string): Promise<void> {
```

Comment:
Instead of batching a single request, after some research, I found a better endpoint.

```http
POST https://contoso.sharepoint.com/sites/project-x/_api/site/RecycleBin/DeleteByIds
{
    "ids": [
        "f4f01b3b-1381-4e0e-a94d-cc2a40f67599",
        "a5f2bc96-ed95-4067-a149-597168b472a9",
        "74645fd5-66e0-4bd1-b983-dbb283794852",
        "6614c74f-0b00-4134-8415-c42c8baeaa75"
    ]
}
```

Let's try to check if there's a limit on how many IDs you can provide.

When providing ids that don't exist we are getting a very unclear error message. I suggest we catch it and throw a clearer message to our users.